vspace/src/main/java/edu/asu/diging/vspace/core/services/ILinkManager.java

```diff
@@ -10,18 +10,19 @@
 import edu.asu.diging.vspace.core.model.display.DisplayType;
 import edu.asu.diging.vspace.core.model.display.ILinkDisplay;
 
-public interface ILinkManager<L extends ILink<T>,T extends IVSpaceElement, U extends ILinkDisplay> {
+public interface ILinkManager<L extends ILink<T>, T extends IVSpaceElement, U extends ILinkDisplay> {
 
-    U createLink(String title, String id, float positionX, float positionY,
-            int rotation, String linkedId, String linkLabel, DisplayType displayType, byte[] linkImage,
-            String imageFilename) throws SpaceDoesNotExistException,ImageCouldNotBeStoredException, SpaceDoesNotExistException;
+    U createLink(String title, String id, float positionX, float positionY, int rotation, String linkedId,
+            String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename, String howToOpen)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException;
 
-    U updateLink(String title, String id, float positionX, float positionY,
-            int rotation, String linkedId, String linkLabel, String linkId, String linkDisplayId,
-            DisplayType displayType, byte[] linkImage, String imageFilename) throws SpaceDoesNotExistException, LinkDoesNotExistsException, ImageCouldNotBeStoredException;
+    U updateLink(String title, String id, float positionX, float positionY, int rotation, String linkedId,
+            String linkLabel, String linkId, String linkDisplayId, DisplayType displayType, byte[] linkImage,
+            String imageFilename, String howToOpen)
+            throws SpaceDoesNotExistException, LinkDoesNotExistsException, ImageCouldNotBeStoredException;
```

Comment:
only external links should open in a new tab by default. I would not change this method as it affects all links but overload it in the subclass.